flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -883,16 +883,7 @@ private void configureCheckpointing() {
 
 		CheckpointingMode mode = cfg.getCheckpointingMode();
 
-		boolean isExactlyOnce;
-		if (mode == CheckpointingMode.EXACTLY_ONCE) {
-			isExactlyOnce = true;
-		} else if (mode == CheckpointingMode.AT_LEAST_ONCE) {
-			isExactlyOnce = false;
-		} else {
-			throw new IllegalStateException("Unexpected checkpointing mode. " +
-				"Did not expect there to be another checkpointing mode besides " +
-				"exactly-once or at-least-once.");
-		}
+		boolean isExactlyOnce = cfg.isCheckpointingEnabled() && mode == CheckpointingMode.EXACTLY_ONCE;
```

Comment:
this change worths a separate hotfix commit

Reply:
I do something further in this round of fixup.

Reply:
I move this to another PR.